flex_config/utils.py

```diff
@@ -0,0 +1,16 @@
+from enum import Enum
+from typing import Any, Dict, List
+
+
+def insert_value_at_nested_key(dest_dict: Dict[str, Any], subkey_path: List[str], value: Any) -> None:
```

Comment:
Same thing here, return the modified dict instead of just mutating for better readability.